src/HierarchicalHandlerInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_list_pages;
+
+/**
+ * Defines an interface for list pages filters hierarchical handlers.
+ */
+interface HierarchicalHandlerInterface {
+
+  /**
+   * Gets the hierarchy for a value.
+   *
+   * @param string $value
+   *   The value.
+   *
+   * @return array
+   *   The hierarchy.
+   */
+  public function getHierarchy(string $value);
```

Comment:
Is there any defined structure this hierarchy must be shaped like? At the minimum, describe this in the comment return.